src/main/java/teammates/ui/webapi/GetHasResponsesAction.java

```diff
@@ -55,11 +59,21 @@ void checkSpecificAccessControl() {
         } else {
             //An student can check whether he has submitted responses for a feedback session in his course.
             String courseId = getNonNullRequestParamValue(Const.ParamsNames.COURSE_ID);
-            String feedbackSessionName = getNonNullRequestParamValue(Const.ParamsNames.FEEDBACK_SESSION_NAME);
-
-            gateKeeper.verifyAccessible(
-                    logic.getStudentForGoogleId(courseId, userInfo.getId()),
-                    getNonNullFeedbackSession(feedbackSessionName, courseId));
+            String feedbackSessionName = getRequestParamValue(Const.ParamsNames.FEEDBACK_SESSION_NAME);
+            if (feedbackSessionName == null) {
+                List<FeedbackSessionAttributes> feedbackSessions = logic.getFeedbackSessionsForCourse(courseId);
+                if (feedbackSessions.isEmpty()) {
+                    // Course has no sessions and therefore no response; access to responses is safe for all.
+                    return;
+                }
+                gateKeeper.verifyAccessible(
+                        logic.getStudentForGoogleId(courseId, userInfo.getId()),
+                        feedbackSessions.get(0));
```

Comment:
why do we only check access for the first session?

Reply:
Good catch; I suppose we need to verify for all sessions indeed.